src/components/Tooltip/Tooltip.js

```diff
@@ -384,6 +394,37 @@ export default class Tooltip extends Component {
           'aria-owns': tooltipId,
         };
 
+    const finalIcon = IconCustomElement ? (
+      <div
```

Comment:
Wondering why you are using a wrapper `<div>` here... In other words, can it be:

```javascript
<IconCustomElement
  name={iconName}
  aria-labelledby={triggerId}
  aria-label={iconDescription}
  ref={node => {
    this.triggerEl = node;
  }}
/>
```

Reply:
Hi @asudoh , with that approach I got a warning about `refs` with stateless components:

`Warning: Stateless function components cannot be given refs. Attempts to access this ref will fail.` 

Also, for some reason if a wrapper is not provided the element duplicates and nests itself, breaking the position of the tooltip. I didn't understand this behavior and tried everything (even `React.Fragment` as wrapper) :

![image](https://user-images.githubusercontent.com/1231888/52154301-3b408e80-264b-11e9-9a0c-205e18fd3e44.png)

The given class name helps to determine if a custom icon is provided.

Reply:
OK thank you for explaining @devnieL! For the warning the user should provide a component with `React.forwardRef()`. For the positioning do you have a reduced case so I can see?

Reply:
Hi @asudoh, sorry, I got confused, I was always passing the `props` argument to the content of `renderIcon` reason because I saw some duplicated attributes, but now I fixed that. 

In respect of the positioning, it got resolved by using the `React.forwardRef()`.

So, will it be required to always use something like this ? 

```js
<Tooltip renderIcon={React.forwardRef(() => <Icon name='icon--add' />)} />
```

I think that's kind of tricky, also it's playing the role of a wrapper that could be considered inside the Tooltip component, maybe there is a way to use React.forwardRef internally when grabbing the `renderIcon` value.

Reply:
I'm looking for the proper way to valid the type of a component with `React.forwardRef`because it returns an object and not a function.

Reply:
@devnieL To better explain this - We need a DOM node of icon, so that we can [calculate the position of icon in the viewport](https://github.com/IBM/carbon-components-react/blob/v6.39.0/src/components/Tooltip/Tooltip.js#L225-L230). My reference to `React.forwardRef()` was too specific and I agree it's confusing, my apologies.

So, the answer is, you need to provide a component where `ref` prop works. Wrapping it with `<div>` makes `ref` prop works automatically, but it causes redundant DOM structure - Such technique should be at user-land therefore.

Reply:
Thanks for the explanation. If we are using refs then the render-function will always result in the warning about ` Stateless function components cannot be given refs.`. So, to confirm, React.forwardRef() will be always required, right ?

```js 
<Tooltip renderIcon={React.forwardRef((props, ref) => <Icon name='icon--add' ref={ref} />)} />
```

Reply:
Makes sense, thank you for double-checking!

Reply:
Also - Please see IBM/carbon-elements#231. Once you update `@carbon/icons-react` dependency in our `package.json` to the latest and re-run `yarn install`, you can `import OverflowMenuVertical16 from '@carbon/icons-react/lib/overflow-menu--vertical/16'` in `Tooltip-story`.js and put `OverflowMenuVertical16` directory to `renderIcon`. That said, I'd recommend making such update throughout `Tooltip-story.js`, or create a similar forward-ref'ed icon component and use it.

Reply:
done, the latest `OverflowMenuVertical16` is wrapped internally with `React.forwardRef` so it can be use directly, thanks!

Reply:
@asudoh with the latest version of `@carbon/icons-react` there is a problem with the CircleCI and netlify checks. how can it be solved ?